daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/DFDLDefineFormat.scala

```diff
@@ -35,8 +35,10 @@ final class DFDLDefineFormat(node: Node, sd: SchemaDocument)
   {
 
   lazy val formatAnnotation = LV(Symbol("formatAnnotation")) {
-    XMLUtils.removeComments(Utility.trim(node)) match {
-      case <defineFormat>{f @ <format>{_*}</format>}</defineFormat> =>
+    val cleanedNode = XMLUtils.removeComments(Utility.trim(node))
```

Comment:
This could be:
```
XMLUtils.removeComments(Utility.trim(node)) match {
   case Elem(_, "defineFormat", _, _, Seq(f @ Elem(_, "format", _, _, _))) => DFDLFormat(f, sd)
   case _ => ...
```
Even in this slightly more complex case, it's easier to do with the pattern matches and the code is far less cluttered.